raiden/raiden_service.py

```diff
@@ -1060,18 +1058,23 @@ def start_mediated_transfer_with_secret(
         fee: FeeAmount,
         target: TargetAddress,
         identifier: PaymentID,
-        secret: Secret,
-        secrethash: SecretHash = None,
+        secret: Optional[Secret],
+        secrethash: Optional[SecretHash] = None,
     ) -> PaymentStatus:
 
+        if secret is None:
+            raise InvalidSecret("No secret is provided. This usage is not supported yet.")
```

Comment:
Offer says Raiden already supports transfers with only secrethash.  So These two lines above should be removed.